File: Examples/ObjectiveC/MoPubBannerAdapterExample/MoPubBannerAdapterExample/ViewController.m
/*
 * Version for iOS © 2015–2021 YANDEX
 *
 * You may not use this file except in compliance with the License.
 * You may obtain a copy of the License at https://yandex.com/legal/mobileads_sdk_agreement/
 */

#import "ViewController.h"
#import <mopub-ios-sdk/MoPub.h>

@interface ViewController () <MPAdViewDelegate>

@property (nonatomic, strong) MPAdView *adView;

@end

@implementation ViewController

- (void)viewDidLoad
{
    [super viewDidLoad];
    
    // Replace XXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXX with Ad Unit ID generated at https://app.mopub.com.
    NSString *adUnit = @"XXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXX";
    self.adView = [[MPAdView alloc] initWithAdUnitId:adUnit];
    self.adView.frame = CGRectMake(0.f, 0.f, 320.f, 50.f);
    self.adView.delegate = self;
    self.adView.translatesAutoresizingMaskIntoConstraints = NO;

    __typeof(self) __weak weakSelf = self;
    MPMoPubConfiguration *configuration = [[MPMoPubConfiguration alloc] initWithAdUnitIdForAppInitialization:adUnit];
    [[MoPub sharedInstance] initializeSdkWithConfiguration:configuration completion:^{
        dispatch_async(dispatch_get_main_queue(), ^{
            [weakSelf.adView loadAd];
        });
    }];
}

- (UIViewController *)viewControllerForPresentingModalView
{
    return self;
}

- (void)adViewDidLoadAd:(MPAdView *)view adSize:(CGSize)adSize
{
    NSLog(@"Ad loaded.");
    [self.adView removeFromSuperview];
    [self.view addSubview:view];
    if (@available(iOS 11.0, *)) {
        [self configureLayoutAtBottomOfSafeAreaForView:view adSize:adSize];
    } else {
        [self configureLayoutAtBottomForView:view adSize:adSize];
    }
}

- (void)configureLayoutAtBottomForView:(MPAdView *)view adSize:(CGSize)adSize
{
    NSDictionary *views = NSDictionaryOfVariableBindings(view);
    NSDictionary *metrics = @{
        @"width" : @(adSize.width),
        @"height" : @(adSize.height)
    };
    NSArray *vertical =
        [NSLayoutConstraint constraintsWithVisualFormat:@"V:[view(=height)]|" options:0 metrics:metrics views:views];
    NSArray *horizontal =
        [NSLayoutConstraint constraintsWithVisualFormat:@"H:[view(=width)]" options:0 metrics:metrics views:views];
    NSLayoutConstraint *center = [NSLayoutConstraint constraintWithItem:view
                                                              attribute:NSLayoutAttributeCenterX
                                                              relatedBy:NSLayoutRelationEqual
                                                                 toItem:self.view
                                                              attribute:NSLayoutAttributeCenterX
                                                             multiplier:1.f
                                                               constant:0.f];
    [self.view addConstraints:vertical];
    [self.view addConstraints:horizontal];
    [self.view addConstraint:center];
}

- (void)configureLayoutAtBottomOfSafeAreaForView:(MPAdView *)view adSize:(CGSize)adSize NS_AVAILABLE_IOS(11_0)
{
    UILayoutGuide *guide = self.view.safeAreaLayoutGuide;
    NSArray *constraints = @[
                             [view.heightAnchor constraintEqualToConstant:adSize.height],
                             [view.centerXAnchor constraintEqualToAnchor:guide.centerXAnchor],
                             [view.widthAnchor constraintEqualToConstant:adSize.width],
                             [view.bottomAnchor constraintEqualToAnchor:guide.bottomAnchor]
                             ];
    [NSLayoutConstraint activateConstraints:constraints];
}

- (void)adView:(MPAdView *)view didFailToLoadAdWithError:(NSError *)error
{
    NSLog(@"Ad failed loading.");
}

@end

File: Examples/Swift/MoPubBannerAdapterExample/MoPubBannerAdapterExample/ViewController.swift
/*
 * Version for iOS © 2015–2021 YANDEX
 *
 * You may not use this file except in compliance with the License.
 * You may obtain a copy of the License at https://yandex.com/legal/mobileads_sdk_agreement/
 */

import MoPubSDK

class ViewController: UIViewController {

    // Replace XXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXX with Ad Unit ID generated at https://app.mopub.com.
    let adUnitId = "XXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXX"

    @IBOutlet weak var loadButton: UIButton!

    var adView: MPAdView!

    override func viewDidLoad() {
        super.viewDidLoad()

        adView = makeAdView()
        initializeMoPub()
    }

    @IBAction func loadAd(_ sender: UIButton) {
        adView.loadAd()
    }
    

    private func initializeMoPub() {
        let configuration = MPMoPubConfiguration(adUnitIdForAppInitialization: adUnitId)
        MoPub.sharedInstance().initializeSdk(with: configuration){ [weak self] in
            DispatchQueue.main.async {
                self?.loadButton.isUserInteractionEnabled = true
            }
        }
    }

    private func makeAdView() -> MPAdView {
        let adView: MPAdView = MPAdView(adUnitId: adUnitId)
        adView.frame = CGRect(x: 0, y: 0, width: 320, height: 50)
        adView.delegate = self
        return adView
    }

    private func addAdView(_ banner: UIView, adSize: CGSize) {
        banner.translatesAutoresizingMaskIntoConstraints = false
        banner.removeFromSuperview()
        view.addSubview(banner)
        configureLayoutForView(banner: banner, size: adSize)
    }

    private func configureLayoutForView(banner: UIView, size: CGSize) {
        let constraints = [NSLayoutConstraint(item: banner,
                                              attribute: .height,
                                              relatedBy: .equal,
                                              toItem: nil,
                                              attribute: .height,
                                              multiplier: 1,
                                              constant: size.height),
                           NSLayoutConstraint(item: banner,
                                              attribute: .width,
                                              relatedBy: .equal,
                                              toItem: nil,
                                              attribute: .width,
                                              multiplier: 1,
                                              constant: size.width),
                           NSLayoutConstraint(item: banner,
                                              attribute: .bottom,
                                              relatedBy: .equal,
                                              toItem: bottomLayoutGuide,
                                              attribute: .top,
                                              multiplier: 1,
                                              constant: 0),
                           NSLayoutConstraint(item: banner,
                                              attribute: .centerX,
                                              relatedBy: .equal,
                                              toItem: view,
                                              attribute: .centerX,
                                              multiplier: 1,
                                              constant: 0)]
        view.addConstraints(constraints)
    }
}

extension ViewController: MPAdViewDelegate {
    func viewControllerForPresentingModalView() -> UIViewController! {
        return self
    }

    func adViewDidLoadAd(_ view: MPAdView!, adSize: CGSize) {
        print("Ad view did load")
        addAdView(adView, adSize: adSize)
    }

    func adView(_ view: MPAdView!, didFailToLoadAdWithError error: Error!) {
        print("Ad view did fail to load ad with error: \(String(describing: error))")
    }
}
